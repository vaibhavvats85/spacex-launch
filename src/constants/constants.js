export const APP_TITLE = 'SpaceX Launch Programs';
export const DEVELOPER_NAME = 'Vaibhav Vats';
export const DEVELOPED_BY = 'Developed by';
export const MISSION_IDS = 'Mission Ids';
export const LAUNCH_YEAR = 'Launch Year';
export const SUCCESSFULL_LAUNCH = 'Successful Launch';
export const SUCCESSFULL_LANDING = 'Successful Landing';
export const FILTERS = 'Filters';
export const LAUNCH_ERROR = 'Unable to Load Launch Programs. We will get bakc to you soon.';
export const NO_PROGRAM = 'No programs available for the given Criteria';
export const LAUNCH_YEAR_KEY = 'launch_year';
export const LAUNCH_SUCCESS_KEY = 'launch_success';
export const LAND_SUCCESS_KEY = 'land_success';
