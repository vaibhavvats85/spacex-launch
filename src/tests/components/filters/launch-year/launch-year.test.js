import { shallow } from 'enzyme';
import LaunchYearFilters from '../../../../components/filters/type/launch-year';
import * as constants from '../../../../constants';
import React from 'react';

let wrapper;
let props = {
    type: constants.LAUNCH_YEAR_KEY,
    clickEvent: jest.fn()
}
jest.mock('react-router-dom', () => ({
    useLocation: () => ({ search: '?launch_year=2006' }),
}));

beforeEach(() => {
    wrapper = shallow(<LaunchYearFilters {...props} />);
});

test('should render LaunchYearFilters component correctly', () => {
    expect(wrapper).toMatchSnapshot();
});


test('should check button correctly', () => {
    const active = wrapper.find('Button').at(0).prop('active').toLowerCase();
    const id = wrapper.find('Button').at(0).prop('value').id.toLowerCase();
    expect(id).toEqual(active)
});


test('should call handleClick event', () => {
    wrapper.find('Button').at(0).prop('handleClick')();
    expect(props.clickEvent).toHaveBeenCalled();
});
