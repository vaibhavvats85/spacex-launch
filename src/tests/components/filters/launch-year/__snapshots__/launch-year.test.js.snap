// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LaunchYearFilters component correctly 1`] = `
Array [
  <Button
    active="2006"
    group="launch_year"
    handleClick={[Function]}
    key="0"
    value={
      Object {
        "id": "2006",
        "val": "2006",
      }
    }
  />,
  <Button
    active="2006"
    group="launch_year"
    handleClick={[Function]}
    key="1"
    value={
      Object {
        "id": "2007",
        "val": "2007",
      }
    }
  />,
  <Button
    active="2006"
    group="launch_year"
    handleClick={[Function]}
    key="2"
    value={
      Object {
        "id": "2008",
        "val": "2008",
      }
    }
  />,
  <Button
    active="2006"
    group="launch_year"
    handleClick={[Function]}
    key="3"
    value={
      Object {
        "id": "2009",
        "val": "2009",
      }
    }
  />,
  <Button
    active="2006"
    group="launch_year"
    handleClick={[Function]}
    key="4"
    value={
      Object {
        "id": "2010",
        "val": "2010",
      }
    }
  />,
  <Button
    active="2006"
    group="launch_year"
    handleClick={[Function]}
    key="5"
    value={
      Object {
        "id": "2011",
        "val": "2011",
      }
    }
  />,
  <Button
    active="2006"
    group="launch_year"
    handleClick={[Function]}
    key="6"
    value={
      Object {
        "id": "2012",
        "val": "2012",
      }
    }
  />,
  <Button
    active="2006"
    group="launch_year"
    handleClick={[Function]}
    key="7"
    value={
      Object {
        "id": "2013",
        "val": "2013",
      }
    }
  />,
  <Button
    active="2006"
    group="launch_year"
    handleClick={[Function]}
    key="8"
    value={
      Object {
        "id": "2014",
        "val": "2014",
      }
    }
  />,
  <Button
    active="2006"
    group="launch_year"
    handleClick={[Function]}
    key="9"
    value={
      Object {
        "id": "2015",
        "val": "2015",
      }
    }
  />,
  <Button
    active="2006"
    group="launch_year"
    handleClick={[Function]}
    key="10"
    value={
      Object {
        "id": "2016",
        "val": "2016",
      }
    }
  />,
  <Button
    active="2006"
    group="launch_year"
    handleClick={[Function]}
    key="11"
    value={
      Object {
        "id": "2017",
        "val": "2017",
      }
    }
  />,
  <Button
    active="2006"
    group="launch_year"
    handleClick={[Function]}
    key="12"
    value={
      Object {
        "id": "2018",
        "val": "2018",
      }
    }
  />,
  <Button
    active="2006"
    group="launch_year"
    handleClick={[Function]}
    key="13"
    value={
      Object {
        "id": "2019",
        "val": "2019",
      }
    }
  />,
  <Button
    active="2006"
    group="launch_year"
    handleClick={[Function]}
    key="14"
    value={
      Object {
        "id": "2020",
        "val": "2020",
      }
    }
  />,
]
`;
